ngraph/core/src/descriptor/tensor.cpp

```diff
@@ -82,6 +72,15 @@ const Shape& descriptor::Tensor::get_shape() const
 {
     if (m_partial_shape.is_static())
     {
+        if (m_shape_changed)
```

Comment:
```suggestion
        if (m_shape_changed.load(std::memory_order_relaxed))
```
This is fastest way to get atomic value from cache if available. In case that we use strong `if(m_shape_changed)` later, it is ok to use 'relaxed' version of load atomic value.